release-content/0.16/release-notes/gpu_driven_rendering.md

```diff
@@ -0,0 +1,75 @@
+
+* *Multi-draw indirect* (MDI), a GPU API that allows multiple meshes to be drawn in a single drawcall, the details of which the GPU provides by filling out tables in GPU memory. In order to use MDI effectively, Bevy uses a new subsystem, the *mesh allocator*, which manages the details of packing meshes together in GPU memory.
+
+  - *Multi-draw indirect count* (MDIC), an extension to multi-draw indirect that allows the GPU to determine the *number* of meshes to draw with minimal overhead.
+
+* *Bindless resources*, which allow Bevy to supply the textures (and other resources) for many objects as a group, instead of having to bind textures one-by-one on the CPU. These resources are managed by a new subsystem known as the *material allocator*.
+
+* *GPU transform and cull*, which allows Bevy to compute the position and visibility of every object from the camera's perspective on the GPU instead of on the GPU.
+
+* The *retained render world*, which allows the CPU to avoid processing and uploading data that hasn't changed since the last frame. Bevy 0.16 not only retains entities in the render world from frame to frame but also retains many internal rendering data structures, allowing it to avoid most CPU overhead associated with meshes that didn't change from the previous frame.
+
+Not all platforms offer full support for GPU-driven rendering; for example, Metal (macOS and iOS) currently only has partial support, and WebGL 2 (the browser) has no support for GPU-driven rendering at all. Some platforms, such as Android phones with certain Mali chips, theoretically support GPU-driven rendering, but in practice driver issues prevent it from working reliably. In these cases, Bevy 0.16 automatically works around the missing functionality, or falls back to CPU-driven rendering if that isn't feasible. This is possible because the Bevy renderer is a unified codebase that supports both CPU- and GPU-driven rendering techniques. That is, there is just one renderer, not a CPU-driven renderer and a GPU-driven renderer.
+
+In most cases, you don't need to do anything special in order for your application to support GPU-driven rendering. There are two main exceptions:
+
+1. Materials with custom WGSL shaders will continue to use CPU-driven rendering by default. In order for your materials to use GPU-driven rendering, you'll want to use the new `#[bindless]` feature on `AsBindGroup`. See the `AsBindGroup` documentation and the `shader_material_bindless` example for more details. If you're using `ExtendedMaterial`, check out the new `extended_material_bindless` example.
+
+2. Applications and plugins that hook into the renderer at a low level will need to be updated to support GPU-driven rendering. The newly-updated `custom_phase_item` and `specialized_mesh_pipeline` examples may prove useful as a guide to do this.
+
+Bevy's current GPU-driven rendering isn't the end of the story. There's a sizable amount of potential future work to be done:
```

Comment:
I'd like us to add a bullet point about leveraging various parts of this to avoid rebinding for CPU-driven too. Say that WebGPU got bindless before MDI, then bindless would be valuable for CPU-driven. The mesh allocator was in 0.15, but could be referenced as something that was already done if interesting to do so.

Reply:
Done.

Reply:
Thanks!!!